### ISSUE TITLE
Repartitioning dask dataframe with unknown divisions

### ISSUE THREAD

**Issue by brunompacheco** [dask/dask#5078]
I need to repartition a dataset given specific divisions and this dataset has unknown divisions. Even though I respect the index boundaries, a error is raised once the operation depends on the old divisions.

```
>>> ddf = dd.read_csv('example.txt')
>>> ddf.head()
   A  B
0  2  5
1  3  6
2  4  7
3  5  8
4  6  9
>>> ddf.set_index('A')
Dask DataFrame Structure:
                   B
npartitions=1
2              int64
7                ...
Dask Name: sort_index, 5 tasks
>>> ddf.known_divisions
False
>>> new_divisions = [2, 4, 7]
>>> ddf.repartition(divisions=new_divisions, force=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python36\lib\site-packages\dask\dataframe\core.py", line 1004, in repartition
    return repartition(self, divisions, force=force)
  File "C:\Program Files\Python36\lib\site-packages\dask\dataframe\core.py", line 4486, in repartition
    df._name, tmp, out, force=force)
  File "C:\Program Files\Python36\lib\site-packages\dask\dataframe\core.py", line 4257, in repartition_divisions
    if a[0] < b[0]:
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

[example.txt](https://github.com/dask/dask/files/3369135/example.txt)

Do I need to map the curent divisions before repartitioning? If so, shouldn't this be verified on the first lines of the function?

**Comment by mrocklin** [comment]
The set_index method is not mutable.  You need to save the result

```diff
- ddf.set_index('A')
+ ddf = ddf.set_index('A')
```

**Comment by brunompacheco** [comment]
Yeah, sorry for that typo. I was trying to come up with a simple example for my situation, in which I have a dataframe with the index set but unknown divisions. I got to that by concatenating several parquets (previously indexed) so the resulting dataframe has the index set, but unknonwn divisions, even though the parquets are ordered.

I took a deeper look and found out that the problem is in the data acquisition: the time reference is messed up for some of the files generated, so the partitions end up overlapping.

But thank you for the quick answer!
